### ISSUE TITLE
Button component - Figma library

### ISSUE THREAD

**Issue by long-chung** [canonical-web-and-design/vanilla-framework#4139]

**Comment by elioqoshi** [comment]
@long-chung are you sure this is a 2? Seems a bit more complex.

Also, should we only have the option to have buttons with margins or also without?

**Comment by long-chung** [comment]
@elioqoshi in terms of level of effort 2 is about right. Mostly copy and pasting.

What other option of buttons should there be?

**Comment by elioqoshi** [comment]
@long-chung I mean whether we want to allow users to have the button without a bounding box that goes beyond the button itself.

<img width="545" alt="image" src="https://user-images.githubusercontent.com/5436686/143023057-ed5d4c45-612d-4ddb-bcae-7b14fc01c2c6.png">

**Comment by long-chung** [comment]
@elioqoshi Yes I agree... However I think having 3 variations of margin for the same button is going to be super confusing for the end user. 
So the question is do we need a button/s with predefined margins @lyubomir-popov

**Comment by elioqoshi** [comment]
I think it's rather a question of 1. whether it makes maintaining the library harder for us and 2. offering too many options creates space for more errors (however we might want to balance that with the fact that if designers get frustrated with component instances, they will end up detaching them and that's worse I think).

For what's worth, one could remove the margins manually in the instance's auto layout if needed.

**Comment by long-chung** [comment]
Ready for review. Thanks

**Comment by elioqoshi** [comment]
Thanks @long-chung. Generally looks good to me. There is some property artifact `¬` in the icon-right property which I removed, it was like this before:

<img width="365" alt="image" src="https://user-images.githubusercontent.com/5436686/143046489-513ef71d-b063-4e62-8d4d-8b994629bb88.png">

I also changed the Type property values to not mention `button` as that's redundant.

Generally I am not sure if it's best practice to have buttons with margins exclusively as a component (I have never seen it in any other design system before), but that's something @lyubomir-popov and I should discuss further.
